uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Wrapper/LeaveRequestDates.php

```diff
@@ -0,0 +1,143 @@
+    return $apiRequest['entity'] === 'LeaveRequest' &&
+           ($apiRequest['action'] === 'get' || $apiRequest['action'] === 'getfull');
+  }
+
+  /**
+   * If the query contains values for the from_date and to_date fields, this
+   * method makes sure they will always contain information about the hour.
+   *
+   * For the from_date field, it will set the hour as 00:00:00 if it's not
+   * present. For the to_date field, it will set it as 23:59:59 if it's not
+   * present.
+   *
+   * These are datetime fields, and the reason for this is that if there's a
+   * query like "to_date <= 2018-01-09" it will include all the leave requests
+   * ending on that date, regardless of the hour. Without this, such a query
+   * would be interpreted as "to_date <= 2018-01-09 00:00:00" by the database,
+   * and most of the requests ending on 2018-01-09 would not be included in the
+   * result.
+   *
+   * @param array $apiRequest
+   */
+  private function prepareDates(&$apiRequest) {
+    $params = &$apiRequest['params'];
+
+    $fromDate = $this->getValueFromParams($params, 'from_date');
+
+    if($fromDate && $this->dateIsValid($fromDate) && $this->dateIsMissingHours($fromDate)) {
+      $fromDate = new DateTime($fromDate);
+      $this->setValueOfParam($params,'from_date', $fromDate->format('Y-m-d 00:00:00'));
+    }
+
+    $toDate = $this->getValueFromParams($params, 'to_date');
+
+    if($toDate && $this->dateIsValid($toDate) && $this->dateIsMissingHours($toDate)) {
+      $toDate = new DateTime($toDate);
+      $this->setValueOfParam($params, 'to_date', $toDate->format('Y-m-d 23:59:59'));
+    }
+  }
+
+  private function dateIsValid($date) {
+    $date = date_parse($date);
+
+    return $date['error_count'] === 0 && $date['warning_count'] === 0;
+  }
+
+  /**
+   * Checks if the given $date string contains information about hours
+   *
+   * @param string $date
+   *  A string in any of the formats supported by strtotime()
+   *
+   * @return bool
+   */
+  private function dateIsMissingHours($date) {
+    $date = date_parse($date);
+
+    return $date['hour'] === false ||
+           $date['minute'] === false ||
+           $date['second'] === false;
+  }
+
+  /**
+   * Gets the value of the given $field in the params array.
+   *
+   * This method works for both simple params ($field => $value) and params with
+   * operators ($field => [$operator => $value])
+   *
+   * @param array $params
+   * @param string $field
+   *
+   * @return mixed|null
+   */
+  private function getValueFromParams(&$params, $field) {
+    if(empty($params[$field])) {
+      return null;
+    }
+
+    // When using other operators than =
+    // the param will be an array in this format:
+    // [<operator> => <value>]
+    if(is_array($params[$field])) {
+      return reset($params[$field]);
+    }
+
+    return $params[$field];
+  }
+
+  /**
+   * Sets the value of the given $field to the given $value in this object's
```

Comment:
> in this object's internal $params array.

This sounds like `$params` is a property of this class.

Reply:
Good catch. These were where originally part of `LeaveRequestSelect` and I forgot to update the comment after I moved them here